add host interaction weight to the hostdb

DavidVorick opened the pull request:
This was originally several nicer commits, but my .git folder corrupted somehow :/

The hostdb was already tracking the reliability of hosts, but this reliability was not actually being factored into the scores of the hosts.

Now it is factored in, although siac doesn't display it in the score breakdown I believe, still need a commit to extend siac.

lukechampine commented on modules/renter.go:
```diff
@@ -98,10 +98,10 @@ type HostDBEntry struct {
-	RecentSuccessfulInteractions   uint64 `json:"recentSuccessfulInteractions"`
+	HistoricSuccessfulInteractions uint64 `json:"historicSuccessfulInteractions"`
 	RecentFailedInteractions       uint64 `json:"recentFailedInteractions"`
+	RecentSuccessfulInteractions   uint64 `json:"recentSuccessfulInteractions"`
```
eesh, somehow I missed that these fields are using camelCase instead of our conventional lowercase. As long as these haven't been exposed in the API yet, we could change them without breaking compatibility. Go ignores case when decoding.

lukechampine commented on modules/renter/hostdb/hostentry.go:
```diff
@@ -0,0 +1,115 @@
+func updateHostHistoricInteractions(host *modules.HostDBEntry, bh types.BlockHeight) {
+	passedTime := bh - host.LastHistoricUpdate
+	if passedTime == 0 {
+		// no time passed. nothing to do.
```
might need a sanity check for `bh < host.LastHistoricUpdate`

DavidVorick replied:
hmm. If the consensus db reset or something, this could actually cause issues. Good catch

DavidVorick commented:
Test should be fixed now. Also added some `siac` output to display the reliability better. Increased the impact of having low reliability. Changed uint64 to float for better precision around small numbers (which they typically are).

This will help with the download and download-repair issues that I'm seeing, but it's not a complete fix.

I think a complete fix will involve making the download code smart enough to prefer hosts that are cheaper and have strong download track records. There's a TODO spot in the codebase to allow this, so I don't think it'll be too much effort to add.
